File: 1394.find-lucky-integer-in-an-array/solution.go
func findLucky(arr []int) int {
    freq := map[int]int{}
    for _, v := range arr {
        freq[v]++
    }
    maxLucky := -1
    for _, v := range arr {
        if v == freq[v] {
            if v > maxLucky {
                maxLucky = v
            }
        }
    }
    return maxLucky
}

File: 1394.find-lucky-integer-in-an-array/solution.kt
class Solution {
    fun findLucky(arr: IntArray): Int {
        return arr
            .asSequence()
            .groupingBy { it }
            .eachCount()
            .filter { it.key == it.value }
            .maxByOrNull { it.key }
            ?.key
        ?: -1 
    }
}